src/test/java/bio/terra/service/auth/iam/sam/SamIamTest.java

```diff
@@ -525,242 +296,493 @@ public void testCreateSnapshotResourceRequestWithPolicySpecifications() throws A
+                IamResourceType.DATASNAPSHOT.getSamResourceName()))
+            .thenReturn(
+                List.of(
+                    new UserResourcesResponse()
+                        .resourceId(id.toString())
+                        .direct(new RolesAndActions().roles(List.of(IamRole.OWNER.toString()))),
+                    new UserResourcesResponse()
+                        .resourceId(id.toString())
+                        .direct(new RolesAndActions().roles(List.of(IamRole.READER.toString())))));
+        Map<UUID, Set<IamRole>> uuidSetMap =
+            samIam.listAuthorizedResources(TEST_USER, IamResourceType.DATASNAPSHOT);
+        assertThat(uuidSetMap, is((Map.of(id, Set.of(IamRole.OWNER, IamRole.READER)))));
+      }
+
+      @Test
+      void listAuthorizedResourcesTest401Error() throws Exception {
+        when(samResourceApi.listResourcesAndPoliciesV2(
+                IamResourceType.DATASNAPSHOT.getSamResourceName()))
+            .thenThrow(IamUnauthorizedException.class);
+        assertThrows(
+            IamUnauthorizedException.class,
+            () -> {
+              samIam.listAuthorizedResources(TEST_USER, IamResourceType.DATASNAPSHOT);
+            });
+      }
+    }
 
-  @Test
-  public void testDeleteGroup() throws ApiException, InterruptedException {
-    String accessToken = TEST_USER.getToken();
-    String groupName = "firecloud_group_name";
+    @Nested
+    class TestSamStatusApi {
+
+      @BeforeEach
+      void setUp() throws Exception {
+        when(samApiService.statusApi()).thenReturn(samStatusApi);
+      }
+
+      @Test
+      void testGetStatus() throws ApiException {
+        when(samStatusApi.getSystemStatus())
+            .thenReturn(
+                new SystemStatus()
+                    .ok(true)
+                    .systems(Map.of("GooglePubSub", new SubsystemStatus().ok(true))));
+        assertThat(
+            samIam.samStatus(),
+            is(
+                new RepositoryStatusModelSystems()
+                    .ok(true)
+                    .message(
+                        """
+                            {GooglePubSub=class SubsystemStatus {
+                                ok: true
+                                messages: null
+                            }}""")));
+      }
+
+      @Test
+      void testGetStatusException() throws ApiException {
+        when(samStatusApi.getSystemStatus()).thenThrow(new ApiException("BOOM!"));
+        assertThat(
+            samIam.samStatus(),
+            is(
+                new RepositoryStatusModelSystems()
+                    .ok(false)
+                    .message(
+                        "Sam status check failed: bio.terra.service.auth.iam.exception.IamInternalServerErrorException: "
+                            + "BOOM!")));
+      }
+    }
 
-    samIam.deleteGroup(accessToken, groupName);
-    verify(samGroupApi).deleteGroup(groupName);
-  }
+    @Nested
+    class TestSamGroupApi {
```

Comment:
Nice, and I think you can also move the different mocks so they're local to the classes. In this case you could move
```
      @Mock private GroupApi samGroupApi;
```
inside this class. Regarding the nested class names, I guess it doesn't really matter although since the outer class (namespace) already has `sam` and `test` in it maybe this could be called `GroupApi` instead of `TestSamGroupApi`?